### ISSUE TITLE
New auth strategy: bearer token

### ISSUE THREAD

**Issue by mfisher87** [nsidc/earthaccess#484]
Related: https://github.com/nsidc/earthaccess/issues/415, https://github.com/nsidc/earthaccess/discussions/479

Should we provide a method of persisting tokens as well? Should bearer token become the default strategy?

**Comment by battistowx** [comment]
This would be awesome to have, instead of creating a requests query to the CMR API

**Comment by fwfichtner** [comment]
First of all thanks for this library, it is very helpful for us! I have a question which is touching this issue. When we are doing `earthaccess.search_data()` we are login in  via [environment variables](https://github.com/nsidc/earthaccess/blob/7a79ea5f1eb752d9050f58e3e0c2813a4409626d/docs/howto/authenticate.md?plain=1#L40). So far so good, under https://urs.earthdata.nasa.gov/users/username/user_tokens we can see that two new tokens are created. I guess these are [CMR Tokens](https://cmr.earthdata.nasa.gov/access-control/site/docs/access-control/api.html)? However, these tokens expire after 2 months and are not re-created and when we continue to use `earthaccess` we are getting `RuntimeError('{"errors":["Token [Bearer eyJXXXI4SIQ] has expired. Note the token value has been partially redacted."]}'`. This is solved by just deleting the Bearer Tokens, which triggers a re-creation.

**Comment by mfisher87** [comment]
Hm. When there are two tokens, the code is supposed to revoke and re-generate a token:

https://github.com/nsidc/earthaccess/blob/7a79ea5f1eb752d9050f58e3e0c2813a4409626d/earthaccess/auth.py#L165-L180

cc @betolink

**Comment by betolink** [comment]
Token regeneration is not automatic, actually is not part of the top level API. It could be automatic if we encounter an expired token, as of now one can do this:

```python
import earthaccess

auth = earthaccess.login()

auth.refresh_tokens()
```

**Comment by mfisher87** [comment]
Ah cool. Looks like we just need a bit of glue code to automate it :) Thanks, @betolink !

**Comment by fwfichtner** [comment]
I had a look to support with a PR, but must admit that I struggle to understand what is going on. 

If I delete the tokens manually they are re-generated, no problem, however and not changing anything I sometimes end up having three working tokens (even though `earthaccess` and the https://urs.earthdata.nasa.gov/users/username/user_tokens expect only two (I can see 3 tokens in the UI). Maybe that happens through some kind of race condition.

<img width="381" alt="Screenshot" src="https://github.com/user-attachments/assets/b5e3b526-d334-4360-979f-1edb7a0035ae">

**Comment by battistowx** [comment]
Would it be better to use [find_or_create_token](https://urs.earthdata.nasa.gov/api/users/find_or_create_token)? We've used that here with no issues: https://github.com/nasa/gesdisc-tutorials/blob/main/notebooks/How_To_Download_a_Spatial_and_Variable_Subset_of_Level_1B_Data_Using_OPeNDAP.ipynb

**Comment by chuckwondo** [comment]
> Would it be better to use [find_or_create_token](https://urs.earthdata.nasa.gov/api/users/find_or_create_token)? We've used that here with no issues: https://github.com/nasa/gesdisc-tutorials/blob/main/notebooks/How_To_Download_a_Spatial_and_Variable_Subset_of_Level_1B_Data_Using_OPeNDAP.ipynb

Yes, indeed, you beat me to the punch.

And in case anybody is wondering, that endpoint is implemented to automatically deal with expired tokens, such that it _always_ returns an active token (automatically generating one and returning it, if you only have expired tokens at the time of the request).

**Comment by fwfichtner** [comment]
Finally found time to check and that would work. Is it okay for you if I create a PR to change the handling of the token?

`_get_user_tokens()` could just use the endpoint you are suggesting. Is there any reason to work with a list of tokens, if not we can also remove the interaction with the other endpoints if `find_or_creat_token` only returns one valid token and also handles re-creation of expired tokens.

**Comment by chuckwondo** [comment]
There should be no need to deal with a list of tokens. Our current token logic should be greatly simplified by making use of `find_or_create_token`. If you want to take a stab at simplifying the logic, go for it. I'll be happy to review your PR.

**Comment by chuckwondo** [comment]
I don't believe that #782 directly addresses this particular issue, so I'm reopening this.

**Comment by mfisher87** [comment]
Should we call this "token-only" strategy for clarity?

**Comment by chuckwondo** [comment]
> Should we call this "token-only" strategy for clarity?

Works for me.
